### PR TITLE
Adding CF annotations to resurrected classes for transitive closure optimization

### PR DESCRIPTION
**Pull request by jurgenvinju**

### CODE REVIEW

**Comment by jurgenvinju** [comment]
If you have a few minutes, this might be fun for you :-)

File: src/main/java/io/usethesource/vallang/impl/util/collections/ShareableValuesHashSet.java [hunk 1]
```diff
@@ -454,55 +335,62 @@ public String toString(){
...
 				return;
 			}
 			
-			for(int i = index - 1; i >= 0 ; i--){
-				Entry<IValue> entry = data[i];
-				if(entry != null){
-					current = entry;
-					index = i;
+			for (int i = this.index - 1; i >= 0 ; i--) {
+				@Nullable Entry<IValue> entry = data[i];
```
Comment (DavyLandman) [inline src/main/java/io/usethesource/vallang/impl/util/collections/ShareableValuesHashSet.java:362]: I think this `@Nullable` is not needed, as locals are always nullable?

File: src/main/java/io/usethesource/vallang/impl/util/collections/ShareableValuesHashSet.java [hunk 2]
```diff
@@ -454,55 +335,62 @@ public String toString(){
...
-			index = 0;
+			this.current = null;
+			this.index = 0;
 		}
 		
-		public boolean hasNext(){
+		@EnsuresNonNullIf(expression="this.current", result=true)
+		@Override
+		public boolean hasNext(@UnknownInitialization SetIterator this) {
 			return (current != null);
```
Comment (DavyLandman) [inline src/main/java/io/usethesource/vallang/impl/util/collections/ShareableValuesHashSet.java:377]: NP: to keep up with the style of the previous function, shouldn't this also be `this.current`?

**Review by DavyLandman — APPROVED** [review]
Few small things, but LGTM

**Pull request merged**